### DISCUSSION TITLE
OpenVPN User Automation

### DISCUSSION THREAD

**Discussion by krvelinov** [jaredhendrickson13/pfsense-api#354] [Q&A]
Hi. 

I am trying to fully automate the OpenVPN Setup from start to finish via the API.

Currently i am able to create the CA Certificate and Server Configuration. However, i am unable to enable the user certificate creation via the API when creating a new User. 

I can see that there is an option for this in the example request:
{
  "authorizedkeys": "",
  "cert": [],
  "descr": "",
  "disabled": false,
  "expires": "",
  "ipsecpsk": "",
  "password": "string",
  "priv": [],
  "username": "string"
}

However, what value should "cert" equal in order for the internal certificate to be created ? 

Furthermore, is there a full documentation on all of the available variables for the API Requests? I might have missed it while checking the git hub, so sorry if i did.

Thank you in advance :)

**Comment by xinspl** [comment]
first you need to create certificate for user and then you can put there its refid.

In System > API > Documantation on evry endpoint you have switch for Schema
